shared/test/showcase/echo/chat_test.rb

```diff
@@ -66,4 +66,33 @@ def test_chat
 
     assert pull_count >= 20, "should have pulled 20 messages by now"
   end
+
+  def test_chat_with_metadata
+    options = Gapic::CallOptions.new metadata: {
+      'showcase-trailer': ["so", "much", "chat"],
+      quiet:              ["please"]
+    }
+    stream_input = Gapic::StreamInput.new
+
+    @client.chat stream_input, options do |response_enum, operation|
+      # TODO: confusing? desirable?
+      assert_nil operation.trailing_metadata
```

Comment:
Ok to be nil here?

Reply:
I don't know how we would avoid it. The call doesn't have trailing metadata until after the call is closed.

I think this is good enough for now. If we come up with a better API then we will change this test to use it.

Reply:
Agreed. I've opened #241 to track it and we'll defer the decision until after we've taken care of higher priority issues.